functions/index.js

```diff
@@ -9,11 +15,83 @@ const functions = require("firebase-functions");
 //  });
 
 // the schedule is specified in cron job format "15 * * * *" = "At minute 15."
-exports.scheduledFunction = functions.region('europe-west1').pubsub.schedule("15 * * * *").onRun((context) => {
+
+
+
+const sections = ["AR", "CGC", "GC", "GM", "EL", "IN", "SV", "MA", "MT", "PH", "MX", "SIE", "SC"]
+
+
+exports.testScheduledFunction = functions.region('europe-west1').pubsub.schedule("15 * * * *").onRun((context) => {
+
+    console.log("It's minute 15!");
   
-  //functions.logger.info("It's minute 15!");
-  console.log("It's minute 15!");
+    const db = getDatabase();
+    const ref = db.ref('Zones');
+
+
+    ref.once('value').then( (snapshot) => {
+
+        console.log("snapshot key " + snapshot.key)
+
+        // update for each zone
+        snapshot.forEach((childSnapshot) => {
+
+            var zone_name = childSnapshot.key;
+            console.log("childSnapshot key " + zone_name.key)
+
+            var owner = "no owner"
+            var max = 0
+            var second_max = 0
+
+            // for each section
+            for(var i = 0, size = sections.length; i < size ; i++){
+
+                var section = sections[i]
+
+                if(childSnapshot.hasChild(section)){
+
+                    let val_section = childSnapshot.child(section).val()
+
+                    console.log(section + " " + val_section)
+                    if(val_section > max){
+                        console.log("val > max")
+                        max = val_section
+                        owner = section
+                    } else if(val_section > second_max){
+                        console.log("val > second_max")
+                        second_max = val_section
```

Comment:
How come you need two max values? To determine if two sections tied...? Makes sense to only take two sections since if more sections tie then only checking for two is enough

Reply:
Yes exactly it is for checking ties